### PR TITLE
Update TypeScript declarations for WebsocketClient

### PR DESCRIPTION
**Pull request by ncallaway**
This updates the TypeScript declarations for the WebsocketClient to match the constructor that's available in javascript.

This is a proposed resolution to Issue #169.

I didn't see any existing mechanism for testing these type declarations, so I didn't add any tests for this. If there is a mechanism for testing type declarations, please let me know and I'll add tests.

All arguments were named to match the names in the javascript WebsocketClient constructor. The only exception is the `channels` argument, which is not specifically named in the javascript as it's an anonymous destructured object. I simply went with the name that seemed to best describe the object itself.

### CODE REVIEW

**Comment by ncallaway** [comment]
I didn't see the other pull request that resolves this issue by iffy here at #160 . If that pull request is accepted, please feel free to close this pull request.

File: index.d.ts [hunk 1]
```diff
@@ -239,8 +239,24 @@ declare module 'gdax' {
         getTrailingVolume(): Promise<any>;
     }
 
+    export interface WebsocketAuthentiation {
```
Comment (wingrunr21) [inline index.d.ts]: You are missing a `c` in `Authentication`

**Comment by wingrunr21** [comment]
Thanks for this! I was just getting ready to fork/PR to fix this.

**Comment by ncallaway** [comment]
Good catch on the spelling! At least it was consistent ;)

**Comment by fb55** [comment]
@mcallaway Sorry, didn’t see this before merging the other one. This seems a bit more complete, would be happy to merge after a rebase!

File: index.d.ts [hunk 2]
```diff
@@ -239,8 +239,24 @@ declare module 'gdax' {
...
+        secret: string,
+        passphrase: string
+    }
+
+    export interface WebsocketChannels {
+        heartbeat?: boolean,
+        channels?: string[],
+    }
+
     export class WebsocketClient {
```
Comment (itajaja) [inline index.d.ts]: shouldn't this extend `EventEmitter`? `@types/node` would have to be installed

**Pull request merged**